llama_index/llms/utils.py

```diff
@@ -6,13 +6,27 @@
 from llama_index.llms.langchain import LangChainLLM
 from llama_index.llms.llama_cpp import LlamaCPP
 from llama_index.llms.llama_utils import messages_to_prompt, completion_to_prompt
+from llama_index.llms.mock import MockLLM
 from llama_index.llms.openai import OpenAI
 
 LLMType = Union[str, LLM, BaseLanguageModel]
 
 
 def resolve_llm(llm: Optional[LLMType] = None) -> LLM:
     """Resolve LLM from string or LLM instance."""
+    if llm == "default":
+        # return default OpenAI model. If it fails, return LlamaCPP
+        try:
+            llm = OpenAI()
+        except ValueError:
+            llm = "local"
+            print(
+                "******\n"
+                "Could not load OpenAI model. Using default LlamaCPP=llama2-13b-chat. "
+                "If you intended to use OpenAI, please check your API key."
```

Comment:
Does this mean that if OPENAI_API_KEY isn't set, then we default to LlamaCPP?

Reply:
Exactly. Just a quick way to unblock people who don't want to set their openai key. This was actually introduced in 0.8.0

Pretty common issue on the discord/github "how do I get rid of OpenAI" -- now you can :)

Reply:
OK, maybe this message should be stronger / a more explicit opt in along with an explicit reference to OPENAI_API_KEY (is there any equivalent to a press enter to continue in notebooks?) I'm not sure if your average person will have the compute/free memory to run Llama2-13B (requires 7GB of free memory) when they are starting out.

Reply:
It runs on CPU, unless you specifically installed llama-cpp compiled for your GPU

I can add a more explicit log though